app/js/app/directives/graph_sketcher/GraphUtils.js

```diff
@@ -271,60 +431,71 @@ define(function() {
+                    potentialPts.push(this.createPoint(pts[i][0], pts[i][1]));
+                }
+            }
 
-            for (let i = 1; i < grad.length; i++) {
-                if (!isNaN(grad[i-1]) && !isNaN(grad[i])) {
-                    if (grad[i] * grad[i-1] < 0 && (pts[i].x - pts[i-1].x) * (pts[i+1].x - pts[i].x) > 0) {
-
-                        let limit = 0.01;
+            let stationaryArrays = Object.create(null);
 
-                        let l = i - 2;
-                        while (l >= 0 && Math.abs(grad[l]) < limit && Math.abs(grad[l]) > Math.abs(grad[l+1]) && grad[l] * grad[l+1] >= 0) {
-                            l--;
-                        }
-                        if (!(Math.abs(grad[l]) >= limit)) {
-                            continue;
-                        }
-
-                        let r = i + 1;
-                        while (r < grad.length && Math.abs(grad[r]) < limit && Math.abs(grad[r]) > Math.abs(grad[r-1]) && grad[r] * grad[r-1] >= 0) {
-                            r++;
-                        }
-                        if (!(Math.abs(grad[r]) >= limit)) {
-                            continue;
-                        }
+            // loop over turn pts and put them in arrays by same y value
+            potentialPts.forEach(function(pt) {
+                let stationaryArray = stationaryArrays[pt[1]];
+                if (!stationaryArray) {
+                    stationaryArray = stationaryArrays[pt[1]] = [];
+                }
+                stationaryArray.push(pt);
+            });
 
-                        let acc1 = grad[l];
-                        let acc2 = grad[r];
-
-                        if (mode == 'maxima') {
-                            if ((pts[i].x > pts[i-1].x && acc1 < 0 && acc2 > 0) || (pts[i].x < pts[i-1].x && acc1 > 0 && acc2 < 0)) {
-                                turnPts.push(this.createPoint(pts[i].x, pts[i].y, pts[i].ind));
-                            } 
-                        } else {
-                            if ((pts[i].x > pts[i-1].x && acc1 > 0 && acc2 < 0) || (pts[i].x < pts[i-1].x && acc1 < 0 && acc2 > 0)) {
-                                turnPts.push(this.createPoint(pts[i].x, pts[i].y, pts[i].ind));
-                            } 
-                        }
+            Object.keys(stationaryArrays).forEach(function(key) {
+                let middle = stationaryArrays[key][Math.floor(stationaryArrays[key].length / 2)];
+                statPts.push(middle);
+            });
 
+            let position = null
 
+            for (let i = 0; i < statPts.length; i++) { 
+                for (let j = 0; j < pts.length; j++) {
+                    if (statPts[i][0] == pts[j][0]) {
+                        position = j;
                     }
                 }
+                ((statPts[i][1] < pts[position-5][1] && statPts[i][1] < pts[position+5][1])) && pot_max.push(statPts[i]);
```

Comment:
This is really an `if` statement, isnt it? There's no need to obscure it to save space; the minifier will do that fine!